app/models/user.rb

```diff
@@ -90,6 +99,22 @@ def missing_required_fields?
 
   private
 
+  def if_driver_remove_unassigned(role)
+    if self.is_a_driver?
+      self.remove_role(:unassigned_driver)
+    end
+  end
+
+  def make_unassigned(role)
+    unless self.is_a_driver?
```

Comment:
I think the passing in of the role happens because these are called by rolify callbacks--after_add and after_remove in these cases, so it's passing in the roles that were added or removed.

As to broadness, I think the bug you caught in .is_a_driver?, where it wasn't scoping tightly enough, limits sufficiently.